packages/frontend/src/providers/ThirdPartyServicesProvider.tsx

```diff
@@ -1,26 +1,70 @@
 import { PostHogProvider, usePostHog } from 'posthog-js/react';
-import { useEffect, type FC } from 'react';
+import { useCallback, useEffect, type FC } from 'react';
 import { IntercomProvider } from 'react-use-intercom';
 import { Intercom } from '../components/Intercom';
 import useSentry from '../hooks/thirdPartyServices/useSentry';
 import useApp from './App/useApp';
 
-const Pylon = () => {
+const usePylon = () => {
     const { user, health } = useApp();
-    useEffect(() => {
-        if (health.data?.pylon?.appId && user.data) {
-            // @ts-ignore
-            window.pylon = {
-                chat_settings: {
-                    app_id: health.data.pylon.appId,
-                    email: user.data.email,
-                    name: `${user.data.firstName} ${user.data.lastName}`,
-                    email_hash: health.data.pylon.verificationHash,
-                },
-            };
-            // @ts-ignore
-            if (window.Pylon) {
+
+    // REFERENCE: https://docs.usepylon.com/pylon-docs/in-app-chat/chat-setup
+    const initPylonWidget = useCallback((appId: string) => {
+        const PYLON_WIDGET_URL = 'https://widget.usepylon.com/widget/';
+        const e = window;
+        const t = document;
+        const n = function () {
+            n.e(arguments); // eslint-disable-line
+        };
+        n.q = [] as unknown[];
+        n.e = function (u: unknown) {
+            n.q.push(u);
+        };
+        // @ts-ignore
+        e.Pylon = n;
+        const r = function () {
+            const s = t.createElement('script');
+            s.setAttribute('type', 'text/javascript');
+            s.setAttribute('async', 'true');
+            s.setAttribute('src', `${PYLON_WIDGET_URL}${appId}`);
+            const ns = t.getElementsByTagName('script')[0];
+            if (ns.parentNode) {
+                ns.parentNode.insertBefore(s, ns);
+            }
+        };
+        if (t.readyState === 'complete') {
+            r();
+        } else if (e.addEventListener) {
+            e.addEventListener('load', r, false);
+        }
+    }, []);
+
+    useEffect(
+        function initPylon() {
+            if (health.data?.pylon?.appId && user.data) {
+                // @ts-ignore
+                window.pylon = {
+                    chat_settings: {
+                        app_id: health.data.pylon.appId,
+                        email: user.data.email,
+                        name: `${user.data.firstName} ${user.data.lastName}`,
+                        email_hash: health.data.pylon.verificationHash,
+                    },
+                };
+
+                // @ts-ignore
+                if (!window.Pylon) {
+                    initPylonWidget(health.data.pylon.appId);
+                }
+            }
+        },
+        [user, health, initPylonWidget],
+    );
+
+    useEffect(
+        function setNewIssueCustomFields() {
+            if (health.data?.pylon?.appId && user.data) {
```

Comment:
We should check if `window.Pylon` is defined. And with that fix we can have a single useEffect.